velox/connectors/hive/HiveConnector.cpp

```diff
@@ -67,23 +67,48 @@ namespace {
 static void makeFieldSpecs(
     const std::string& pathPrefix,
     int32_t level,
-    const std::shared_ptr<const RowType>& type,
+    const std::shared_ptr<const Type>& type,
     common::ScanSpec* spec) {
-  for (auto i = 0; i < type->size(); ++i) {
-    std::string path =
-        level == 0 ? type->nameOf(i) : pathPrefix + "." + type->nameOf(i);
-    common::Subfield subfield(path);
-    common::ScanSpec* fieldSpec = spec->getOrCreateChild(subfield);
-    fieldSpec->setProjectOut(true);
-    fieldSpec->setChannel(i);
-    auto fieldType = type->childAt(i);
-    if (fieldType->kind() == TypeKind::ROW) {
+  switch (type->kind()) {
+    case TypeKind::ROW: {
+      auto rowType = dynamic_cast<const RowType*>(type.get());
+      for (auto i = 0; i < type->size(); ++i) {
+        std::string path = level == 0 ? rowType->nameOf(i)
+                                      : pathPrefix + "." + rowType->nameOf(i);
+        common::Subfield subfield(path);
+        common::ScanSpec* fieldSpec = spec->getOrCreateChild(subfield);
+        fieldSpec->setProjectOut(true);
+        fieldSpec->setChannel(i);
+        makeFieldSpecs(path, level + 1, type->childAt(i), spec);
+      }
+      break;
+    }
+    case TypeKind::MAP: {
+      auto keySpec =
+          spec->getOrCreateChild(common::Subfield(pathPrefix + ".keys"));
```

Comment:
nit: introduce temporary "path" variable avoid repeating pathPrefix + ".keys" in multiple places